src/OrchardCore.Modules/OrchardCore.Users/Controllers/ExternalAuthenticationsController.cs

```diff
@@ -520,6 +520,75 @@ public async Task<IActionResult> RemoveLogin(RemoveLoginViewModel model)
         return RedirectToAction(nameof(ExternalLogins));
     }
 
+    private async Task UpdateAndValidatePasswordAsync(RegisterExternalLoginViewModel model, bool noPassword)
```

Comment:
Why are these better here instead of the view model? Note that moving the T-strings here will need the translations to be updated too.

Reply:
To start, I am here using the IdentityOptions to validation the password asynchronous. Using IValidateObject you can't call asynchronous code without locking. Second, the implementation depended on a value set by setting which is wired. Validating in the controller allows for asynchronous calls and also make it clear what we are validating.

Reply:
OK, thanks for explaining.